sirepo/job_supervisor.py

```diff
@@ -304,44 +328,18 @@ async def receive(cls, req):
             pkdlog('{} error={} stack={}', req, e, pkdexc())
             return sirepo.http_reply.gen_tornado_exception(e)
 
-    async def run_dir_acquire(self, owner):
-        if self.run_dir_owner == owner:
+    def set_status(self, op, status, exception=None):
+        if op.opName != job.OP_RUN:
             return
-        e = None
-        if not self.run_dir_mutex.is_set():
-            pkdlog('{} await self.run_dir_mutex', self)
-            await self.run_dir_mutex.wait()
-            e = Awaited()
-            if self.run_dir_owner:
-                # some other op acquired it before this one
-                raise e
-        self.run_dir_mutex.clear()
-        self.run_dir_owner = owner
-        if e:
-            raise e
-
-    def run_dir_release(self, owner):
-        assert owner == self.run_dir_owner, \
-            'owner={} not same as releaser={}'.format(self.run_dir_owner, owner)
-        self.run_dir_owner = None
-        self.run_dir_mutex.set()
-
-    def set_status(self, op, status):
-        m = None
-        if op.opName == job.OP_RUN:
-            if status == _Op.STATUS_AWAIT_OP_SLOT:
-                m = 'Waiting for another simulation to complete'
-            elif status == _Op.STATUS_COMPUTE_RUNNING:
-                m = 'Running'
-
-        if m:
-            assert not self.db.jobStatusMessage, \
-                'Trying to overwrite existing jobStatusMessage={}'.format(
-                    self.db.jobStatusMessage,
-                )
-            self.__db_update(
-                jobStatusMessage=m,
-            )
+        s = self.db.jobStatusMessage
+        p = 'Exception: '
+        if status is not None:
+            # POSIT: no other status begins with exception
+            assert not s or not s.startswith(p), \
+                f'Trying to overwrite existing jobStatusMessage="{s}" with status="{status}"'
+        if exception is not None:
+            status = f'{p}{e}, while {s}'
```

Comment:
i think it should be {exception} not {e}. I don't see e defined anywhere.

Reply:
Good catch. BTW, did you happen to see how status is being set now differently than before? It allows the exception to be cascaded to the jobStatusMessage.

Reply:
374a8cb5

Reply:
No I missed that. Thanks for pointing it out.

Reply:
Do you think it is a good idea, and does the algorithm work correctly?

Reply:
I think you want `{type(exception)}` or something to that effect. Just doing `{e}` prints out the message of the exception. In the case of CancelledError this is empty so we get jobStatusMessage="Exception: , while Waiting for another simulation to complete"

I think it's a good idea. But, I do think we (maybe just me if you have a grasp) should take time and think about what we want to be in the fields  alert, error, statusMessage, and internalError. It may be worth documenting them. They all may contain a message about what went wrong and it isn't immediately clear what should go where and when they should/shouldn't overlap.

For example, if a user cancels a simulation that is waiting on a resource then the jobStatusMessage='exception: cancelledErrror while waiting for ...'., the status=cancelled, and the error=None. The message has 'exception' but error=None. Is that confusing? Not to me because I know how this all works but may be to someone else. In addition currently we don't display the jobStatusMessage to user's for non running/pending sims but if we change that in the future then they will see some info about 'exception ...' even though from their perspective there was no "error" (exceptions are just how we cancel things internally).

Reply:
Thinking about this lead me to uncover that the python folks broke CancelledError in py 3.8. It now  inherits from BaseException instead of Exception. This is going to break us in more than a few places. I sent an email to sw to discuss.

https://bugs.python.org/issue32528